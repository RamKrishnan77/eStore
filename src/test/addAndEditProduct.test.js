import {shallow} from 'enzyme'
import React from 'react'
import AddProduct from '../components/addProduct.component'
import EditProduct from '../components/editProduct.component'
import ProfilePage from '../components/ProfilePage.component'
import MyCard from '../components/card.component'
import CheckBox from '../components/filter.component'
import SearchBar from '../components/searchBar.component'
import CustomButton from '../components/custom-button.component'
import FormInput from '../components/formInput.component'
import ProductDetailsPage from '../components/productDetailsPage.component'
import SearchResults from '../components/searchResults.component'
import store from '../redux/store' 
import {Provider} from 'react-redux'

//Snapshot testing of AddProduct Page

it('expect to render AddProduct Page', () => {
    expect(shallow(<Provider store={store}><AddProduct></AddProduct></Provider>)).toMatchSnapshot()
})

//Snapshot testing of EditProduct Page

it('expect to render EditProduct Page', () => {
    expect(shallow(<Provider store={store}><EditProduct></EditProduct></Provider>)).toMatchSnapshot()
})

//Snapshot testing of Profile Page

it('expect to render Profile Page', () => {
    expect(shallow(<Provider store={store}><ProfilePage></ProfilePage></Provider>)).toMatchSnapshot()
})

//Snapshot testing of Card

it('expect to render Card', () => {
    expect(shallow(<Provider store={store}><MyCard></MyCard></Provider>)).toMatchSnapshot()
})

//Snapshot testing of filter

it('expect to render Filter', () => {
    expect(shallow(<Provider store={store}><CheckBox></CheckBox></Provider>)).toMatchSnapshot()
})

//Snapshot testing of SearchBar

it('expect to render SearchBar', () => {
    expect(shallow(<Provider store={store}><SearchBar></SearchBar></Provider>)).toMatchSnapshot()
})

//Snapshot testing of SearchResults

it('expect to render SearcResults', () => {
    expect(shallow(<Provider store={store}><SearchResults></SearchResults></Provider>)).toMatchSnapshot()
})

//Snapshot testing of ProductDetails

it('expect to render ProductDetails', () => {
    expect(shallow(<Provider store={store}><ProductDetailsPage></ProductDetailsPage></Provider>)).toMatchSnapshot()
})

//Snapshot testing of CustomButton

it('expect to render CustomButton', () => {
    expect(shallow(<Provider store={store}><CustomButton></CustomButton></Provider>)).toMatchSnapshot()
})

//Snapshot testing of formInput

it('expect to render FormInput', () => {
    expect(shallow(<Provider store={store}><FormInput></FormInput></Provider>)).toMatchSnapshot()
})

