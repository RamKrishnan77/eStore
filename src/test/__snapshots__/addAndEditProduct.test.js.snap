// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render AddProduct Page 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(withRouter(AddProduct)) />
</ContextProvider>
`;

exports[`expect to render Card 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(withRouter(MyCard)) />
</ContextProvider>
`;

exports[`expect to render CustomButton 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CustomButton />
</ContextProvider>
`;

exports[`expect to render EditProduct Page 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(withRouter(EditProduct)) />
</ContextProvider>
`;

exports[`expect to render Filter 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CheckBox />
</ContextProvider>
`;

exports[`expect to render FormInput 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FormInput />
</ContextProvider>
`;

exports[`expect to render ProductDetails 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(ProductDetailsPage) />
</ContextProvider>
`;

exports[`expect to render Profile Page 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ProfilePage />
</ContextProvider>
`;

exports[`expect to render SearcResults 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(SearchResults) />
</ContextProvider>
`;

exports[`expect to render SearchBar 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <SearchBar />
</ContextProvider>
`;
